src/App.jsx

```diff
@@ -14,45 +16,105 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_FIELD_NAME = 'name';
+const SORT_FIELD_LENGTH = 'length';
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reverse, setReverse] = useState(false);
+  const [sortedGoods, setSortedGoods] = useState(goodsFromServer);
+
+  const handleAlphabeticalSort = () => {
+    setSortField(SORT_FIELD_NAME);
+    setSortedGoods([...sortedGoods]
+      .sort((good1, good2) => {
+        const sortFunc = reverse
+          ? good2.localeCompare(good1)
+          : good1.localeCompare(good2);
+
+        return sortFunc;
+      }));
+  };
+
+  const handleLengthSort = () => {
+    setSortField(SORT_FIELD_LENGTH);
+    setSortedGoods([...sortedGoods]
+      .sort((good1, good2) => {
+        const sortFunc = reverse
+          ? good2.length - good1.length
+          : good1.length - good2.length;
+
+        return sortFunc;
+      }));
+  };
+
+  const handleRevers = () => {
+    setReverse(!reverse);
+    setSortedGoods([...sortedGoods].reverse());
+  };
+
+  const handleReset = () => {
+    setSortField('');
+    setSortedGoods([...goodsFromServer]);
+    setReverse(false);
+  };
```

Comment:
very good solution to make method for reset 🔥, but without `setSortedGoods`, see comments above